### PR TITLE
SRE-2677 support kubernetes provider in secret store

### PR DESCRIPTION
**Pull request by dansc0de**
helm template output with this test yaml

```yaml
global:
  namespaceOverride: "auth-svc"
  fullnameOverride: "auth-svc"

namespace: {}
serviceName: {}

stores:
  kubernetes:
    type: "kubernetes"
    secretStoreNameOverride: "kafka-ssc"
    remoteNamespace: "kafka-ssc"
    secrets:
      - sourcePath: "kafka-ssc-clients-ca-cert"
        sourceKey: "ca.crt"
      - sourcePath: "kafka-ssc-test-user"
        sourceKey: "user.crt"
      - sourcePath: "kafka-ssc-test-user"
        sourceKey: "user.key"
      - sourcePath: "kafka-ssc-test-user"
        sourceKey: "user.password"
```

Helm template output 

```yaml
---
# Source: secretstores/templates/rbac.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: "kafka-ssc-secretstore-migrate-secrets"
  namespace: "auth-svc"
---
# Source: secretstores/templates/rbac.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: "auth-svc-migrate-secrets-role"
  namespace: "kafka-ssc"
rules:
  - apiGroups: [""]
    resources:
      - secrets
    verbs:
      - get
      - list
      - watch
  - apiGroups:
      - authorization.k8s.io
    resources:
      - selfsubjectrulesreviews
    verbs:
      - create
---
# Source: secretstores/templates/rbac.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: "auth-svc-migrate-secrets-rolebinding"
  namespace: "kafka-ssc"
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: "auth-svc-migrate-secrets-role"
subjects:
  - kind: ServiceAccount
    name: "kafka-ssc-secretstore-migrate-secrets"
    namespace: "auth-svc"
---
# Source: secretstores/templates/externalsecret.yaml
apiVersion: external-secrets.io/v1beta1
kind: ExternalSecret
metadata:
  name: "secrets-kafka-ssc-secretstore"
  namespace: "auth-svc"
  labels:
    helm.sh/chart: "secretstores-0.0.5"
    app.kubernetes.io/name: "secretstores"
    app.kubernetes.io/instance: "auth-svc"
    tags.tatari.tv/env: 
    app.kubernetes.io/version: "0.0.5"
    app.kubernetes.io/managed-by: "Helm"
spec:
  refreshInterval: "1h"
  secretStoreRef:
    name: "kafka-ssc-secretstore"
    kind: "SecretStore"
  target:
    name: "secrets-kafka-ssc-secretstore"
  data:
    - remoteRef:
        key: "kafka-ssc-clients-ca-cert"
        property: "ca.crt"
      secretKey: "ca.crt"
    - remoteRef:
        key: "kafka-ssc-test-user"
        property: "user.crt"
      secretKey: "user.crt"
    - remoteRef:
        key: "kafka-ssc-test-user"
        property: "user.key"
      secretKey: "user.key"
    - remoteRef:
        key: "kafka-ssc-test-user"
        property: "user.password"
      secretKey: "user.password"
---
# Source: secretstores/templates/secretstore.yaml
apiVersion: external-secrets.io/v1beta1
kind: SecretStore
metadata:
  name: "kafka-ssc-secretstore"
  namespace: "auth-svc"
  labels:
    helm.sh/chart: "secretstores-0.0.5"
    app.kubernetes.io/name: "secretstores"
    app.kubernetes.io/instance: "auth-svc"
    tags.tatari.tv/env: 
    app.kubernetes.io/version: "0.0.5"
    app.kubernetes.io/managed-by: "Helm"
spec:
  # Provider field contains the configuration to access the provider which
  # contains the secret.
  # Exactly one provider must be configured.
  provider:
    # Kubernetes
    # Sync secrets across namespaces in the same cluster
    kubernetes:
      # remote namespace to sync secrets from
      remoteNamespace: "kafka-ssc"
      server:
        # @note: force internal cluster usage. We do not want to sync secrets from external clusters
        url: https://kubernetes.default.svc
        caProvider:
          type: ConfigMap
          name: kube-root-ca.crt
          key: ca.crt
      auth:
        serviceAccount:
          name: "kafka-ssc-secretstore-migrate-secrets"
```

### CODE REVIEW

**Review by stephen-tatari — APPROVED** [review]

**Comment by dansc0de** [comment]
@pshelby please let me know what you think about the update change using a different key structure.

File: charts/secretstores/templates/_helpers.tpl [hunk 1]
```diff
@@ -68,3 +68,17 @@ Set namespace by fullname or the environment values override
...
+*/}}
+{{- define "secretstores.storeName" -}}
+{{- $key := .key -}}
+{{- $value := .value -}}
+  {{- if and (eq $value.type "kubernetes") -}}
+    {{- printf "%s-%s" $value.remoteNamespace "secretstore" }}
+  {{- else -}}
+    {{- $key | quote -}}
+  {{- end -}}
+{{- end -}}
```
Comment (pshelby) [inline charts/secretstores/templates/_helpers.tpl:84]: nit: maybe some linting here (multiple newlines, no newline at EOF)?

**Review by pshelby — APPROVED** [review]
Cool!  That looks like it will be more flexible.  🚀

File: charts/secretstores/templates/_helpers.tpl (hunk 1 above)
Comment (dansc0de) [reply charts/secretstores/templates/_helpers.tpl]: good catch it is not used.

**Pull request merged**